First bootstrap raise error dockerflow.health.E001 Could not connect to database

## Bug Report

**Problematic Behavior**
On a clean installation, the bootstrap/run, won't start up the Richie sandbox application.

On an existing installation, to simulate the problem, we need to remove the existing sandbox database and then run it.
```bash
docker rm richie-postgresql-1
make run
```

Logs, 
```
[+] Running 4/0
 ✔ Container richie-elasticsearch-1  Running                                                                                           0.0s
 ✔ Container richie-postgresql-1     Running                                                                                           0.0s
 ✔ Container richie-app-1            Running                                                                                           0.0s
 ✔ Container richie-nginx-1          Running                                                                                           0.0s
Wait for services to be up...
2024/07/02 12:06:57 Waiting for: tcp://postgresql:5432
2024/07/02 12:06:57 Connected to tcp://postgresql:5432
2024/07/02 12:07:01 Waiting for: tcp://elasticsearch:9200
2024/07/02 12:07:01 Connected to tcp://elasticsearch:9200
2024/07/02 12:07:07 Waiting for: tcp://app:8000
2024/07/02 12:07:07 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:08 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:09 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:10 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:11 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:12 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:13 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:14 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:15 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:16 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:17 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:18 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:19 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:20 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:21 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:22 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:23 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:24 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:25 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:26 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:27 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:28 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:29 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:30 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:31 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:32 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:33 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:34 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:35 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:36 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:37 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:38 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:39 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:40 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:41 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:42 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:43 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:44 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:45 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:46 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:47 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:48 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:49 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:50 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:51 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:52 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:53 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:54 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:55 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:56 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:57 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:58 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:07:59 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:08:00 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:08:01 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:08:02 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:08:03 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:08:04 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:08:05 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:08:06 Problem with dial: dial tcp 172.19.0.4:8000: getsockopt: connection refused. Sleeping 1s
2024/07/02 12:08:07 Timeout after 1m0s waiting on dependencies to become available: [tcp://app:8000]
make: *** [Makefile:116: run] Error 1
```

logs
```
$ make logs
app-1  | 🐳(entrypoint) creating user running in the container...
app-1  | 🐳(entrypoint) running your command: /bin/sh -c cd sandbox && python manage.py runserver 0.0.0.0:8000
app-1  | django-configurations version 2.5.1, using configuration Development
app-1  | Performing system checks...
app-1  |
app-1  | Exception in thread django-main-thread:
app-1  | Traceback (most recent call last):
app-1  |   File "/usr/local/lib/python3.11/threading.py", line 1045, in _bootstrap_inner
app-1  |     self.run()
app-1  |   File "/usr/local/lib/python3.11/threading.py", line 982, in run
app-1  |     self._target(*self._args, **self._kwargs)
app-1  |   File "/usr/local/lib/python3.11/site-packages/django/utils/autoreload.py", line 64, in wrapper
app-1  |     fn(*args, **kwargs)
app-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/management/commands/runserver.py", line 133, in inner_run
app-1  |     self.check(display_num_errors=True)
app-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 556, in check
app-1  |     raise SystemCheckError(msg)
app-1  | django.core.management.base.SystemCheckError: SystemCheckError: System check identified some issues:
app-1  |
app-1  | ERRORS:
app-1  | ?: (dockerflow.health.E001) Could not connect to database: connection to server at "postgresql" (172.19.0.2), port 5432 failed: Connection refused
app-1  |        Is the server running on that host and accepting TCP/IP connections?
app-1  |
app-1  |
app-1  | INFOS:
app-1  | ?: (dockerflow.health.I001) Can't connect to database to check migrations
app-1  |
app-1  | System check identified 2 issues (0 silenced).
```

Database logs:
```
$ docker logs richie-postgresql-1
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locale "en_US.utf8".
The default database encoding has accordingly been set to "UTF8".
The default text search configuration will be set to "english".

Data page checksums are disabled.

fixing permissions on existing directory /var/lib/postgresql/data ... ok
creating subdirectories ... ok
selecting dynamic shared memory implementation ... posix
selecting default max_connections ... 100
selecting default shared_buffers ... 128MB
selecting default time zone ... Etc/UTC
creating configuration files ... ok
running bootstrap script ... ok
performing post-bootstrap initialization ... ok
syncing data to disk ... ok


Success. You can now start the database server using:

    pg_ctl -D /var/lib/postgresql/data -l logfile start

initdb: warning: enabling "trust" authentication for local connections
You can change this by editing pg_hba.conf or using the option -A, or
--auth-local and --auth-host, the next time you run initdb.
waiting for server to start....2024-07-02 11:55:12.217 UTC [49] LOG:  starting PostgreSQL 12.19 (Debian 12.19-1.pgdg120+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
2024-07-02 11:55:12.239 UTC [49] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2024-07-02 11:55:12.381 UTC [50] LOG:  database system was shut down at 2024-07-02 11:53:57 UTC
2024-07-02 11:55:12.416 UTC [49] LOG:  database system is ready to accept connections
 done
server started
CREATE DATABASE
```

**Expected behavior/code**
To run it.

**Steps to Reproduce**
1. Clone
2. Execute `make bootstrap`, it is failing on the `run` Makefile target.

**Environment**
- Richie version: master
- Platform: Linux

**Possible Solution**
<!--- Only if you have suggestions on a fix for the bug -->
I think the database isn't ready, when the dockerflow is checking if the database connection.
The waits that we have on the Makefile only check if port is open and not if the database is ready to accept connections.
I think we could get some inspiration on: https://stackoverflow.com/a/55835081/2239848

**Additional context/Screenshots**
Add any other context about the problem here. If applicable, add screenshots to help explain.

--- a/Makefile
+++ b/Makefile
@@ -43,14 +43,13 @@ DOCKER_UID           = $(shell id -u)
 DOCKER_GID           = $(shell id -g)
 DOCKER_USER          = $(DOCKER_UID):$(DOCKER_GID)
 COMPOSE              = DOCKER_USER=$(DOCKER_USER) DB_HOST=$(DB_HOST) DB_PORT=$(DB_PORT) docker compose -f docker-compose.yml -f docker-compose-$(DB_HOST).yml
-COMPOSE_SSL          = NGINX_CONF=ssl DEV_ENV_FILE=dev-ssl $(COMPOSE)
+SSL_ENV              = NGINX_CONF=ssl DEV_ENV_FILE=dev-ssl
 COMPOSE_RUN          = $(COMPOSE) run --rm
-COMPOSE_RUN_SSL      = $(COMPOSE_SSL) run --rm
 COMPOSE_EXEC         = $(COMPOSE) exec
+COMPOSE_UP_WAIT      = $(COMPOSE) up --wait --wait-timeout 60
 COMPOSE_EXEC_APP     = $(COMPOSE_EXEC) app
 COMPOSE_EXEC_FRONTEND= $(COMPOSE_EXEC) frontend
 COMPOSE_RUN_APP      = $(COMPOSE_RUN) app
-COMPOSE_RUN_SSL_APP  = $(COMPOSE_RUN_SSL) app
 COMPOSE_RUN_CROWDIN  = $(COMPOSE_RUN) crowdin crowdin
 COMPOSE_TEST_RUN     = $(COMPOSE) run --rm -e DJANGO_CONFIGURATION=Test
 COMPOSE_TEST_RUN_APP = $(COMPOSE_TEST_RUN) app
@@ -68,10 +67,6 @@ YARN                 = $(COMPOSE_RUN_FRONTEND) yarn
 
 # -- Django
 MANAGE               = $(COMPOSE_RUN_APP) python sandbox/manage.py
-MANAGE_SSL           = $(COMPOSE_RUN_SSL_APP) python sandbox/manage.py
-WAIT_DB              = $(COMPOSE_RUN) dockerize -wait tcp://$(DB_HOST):$(DB_PORT) -timeout 60s
-WAIT_ES              = $(COMPOSE_RUN) dockerize -wait tcp://elasticsearch:9200 -timeout 60s
-WAIT_APP             = $(COMPOSE_RUN) dockerize -wait tcp://app:8000 -timeout 60s
 
 # ==============================================================================
 # RULES
@@ -109,19 +104,13 @@ logs: ## display app logs (follow mode)
 .PHONY: logs
 
 run: ## start the development server
-	@$(COMPOSE) up -d nginx
-	@echo "Wait for services to be up..."
-	@$(WAIT_DB)
-	@$(WAIT_ES)
-	@$(WAIT_APP)
+	@echo "Start and wait for ${DB_HOST}, elasticsearch and app to be up..."
+	@$(COMPOSE_UP_WAIT) -d nginx $(DB_HOST) elasticsearch app
 .PHONY: run
 
 run-ssl: ## start the development server over TLS
-	@$(COMPOSE_SSL) up -d nginx
-	@echo "Wait for services to be up..."
-	@$(WAIT_DB)
-	@$(WAIT_ES)
-	@$(WAIT_APP)
+	@echo "Start and wait for services to be up..."
+	@$(SSL_ENV) $(COMPOSE_UP_WAIT) -d nginx $(DB_HOST) elasticsearch app
 .PHONY: run-ssl
 
 status: ## an alias for "docker compose ps"
@@ -249,23 +238,20 @@ messages: ## create the .po files used for i18n
 .PHONY: messages
 
 migrate: ## perform database migrations
-	@$(COMPOSE) up -d ${DB_HOST}
-	@$(WAIT_DB)
+	@echo "Start and wait for ${DB_HOST} to be up..."
+	@$(COMPOSE_UP_WAIT) -d ${DB_HOST}
 	@$(MANAGE) migrate
 .PHONY: migrate
 
 search-index: ## (re)generate the Elasticsearch index
-	@$(COMPOSE) up -d ${DB_HOST}
-	@$(WAIT_DB)
-	@$(COMPOSE) up -d elasticsearch
-	@$(WAIT_ES)
+	@echo "Start and wait for ${DB_HOST} & elasticsearch to be up..."
+	@$(COMPOSE_UP_WAIT) -d ${DB_HOST} elasticsearch
 	@$(MANAGE) bootstrap_elasticsearch
 .PHONY: search-index
 
 superuser: ## Create an admin user with password "admin"
-	@$(COMPOSE) up -d ${DB_HOST}
-	@echo "Wait for services to be up..."
-	@$(WAIT_DB)
+	@echo "Start and wait for ${DB_HOST} to be up..."
+	@$(COMPOSE_UP_WAIT) -d ${DB_HOST}
 	@$(MANAGE) shell -c "from django.contrib.auth.models import User; not User.objects.filter(username='admin').exists() and User.objects.create_superuser('admin', 'admin@example.com', 'admin')"
 .PHONY: superuser
 

--- a/Makefile
+++ b/Makefile
@@ -42,7 +42,7 @@ endif
 DOCKER_UID           = $(shell id -u)
 DOCKER_GID           = $(shell id -g)
 DOCKER_USER          = $(DOCKER_UID):$(DOCKER_GID)
-COMPOSE              = DOCKER_USER=$(DOCKER_USER) DB_HOST=$(DB_HOST) DB_PORT=$(DB_PORT) docker compose
+COMPOSE              = DOCKER_USER=$(DOCKER_USER) DB_HOST=$(DB_HOST) DB_PORT=$(DB_PORT) docker compose -f docker-compose.yml -f docker-compose-$(DB_HOST).yml
 COMPOSE_SSL          = NGINX_CONF=ssl DEV_ENV_FILE=dev-ssl $(COMPOSE)
 COMPOSE_RUN          = $(COMPOSE) run --rm
 COMPOSE_RUN_SSL      = $(COMPOSE_SSL) run --rm
@@ -263,7 +263,7 @@ search-index: ## (re)generate the Elasticsearch index
 .PHONY: search-index
 
 superuser: ## Create an admin user with password "admin"
-	@$(COMPOSE) up -d mysql
+	@$(COMPOSE) up -d ${DB_HOST}
 	@echo "Wait for services to be up..."
 	@$(WAIT_DB)
 	@$(MANAGE) shell -c "from django.contrib.auth.models import User; not User.objects.filter(username='admin').exists() and User.objects.create_superuser('admin', 'admin@example.com', 'admin')"
@@ -326,7 +326,7 @@ i18n-generate-front: build-ts
 # -- Database
 
 dbshell: ## connect to database shell
-	docker compose exec app python sandbox/manage.py dbshell
+	@$(COMPOSE) exec app python sandbox/manage.py dbshell
 .PHONY: dbshell
 
 # -- Misc

--- a/bin/_config.sh
+++ b/bin/_config.sh
@@ -7,12 +7,8 @@ UNSET_USER=0
 COMPOSE_PROJECT="richie"
 
 # By default, all commands are run on Postgresql.
-# They can be run on Mysql by setting an environment variable: `DB_ENGINE=mysql`
-if [[ "${DB_ENGINE}" == "mysql" ]]; then
-    COMPOSE_FILE="${REPO_DIR}/docker/compose/development/mysql/docker-compose.yml"
-else
-    COMPOSE_FILE="${REPO_DIR}/docker-compose.yml"
-fi
+# They can be run on Mysql by setting an environment variable: `DB_HOST=mysql`
+COMPOSE_FILE="-f ${REPO_DIR}/docker-compose.yml -f ${REPO_DIR}/docker-compose-${DB_HOST:-postgresql}.yml"
 
 # _set_user: set (or unset) default user id used to run docker commands
 #
@@ -47,8 +43,9 @@ function _docker_compose() {
     echo "🐳(compose) project: '${COMPOSE_PROJECT}' file: '${COMPOSE_FILE}'"
     docker compose \
         -p "${COMPOSE_PROJECT}" \
-        -f "${COMPOSE_FILE}" \
+        ${COMPOSE_FILE} \
         --project-directory "${REPO_DIR}" \
+        --env-file env.d/development/${DB_HOST:-postgresql}  \
         "$@"
 }
 

--- a/docker-compose-mysql.yml
+++ b/docker-compose-mysql.yml
@@ -1,0 +1,17 @@
+services:
+  mysql:
+    image: mysql:8.0
+    env_file:
+      - env.d/development/mysql
+    command: --default-authentication-plugin=mysql_native_password
+    healthcheck:
+      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
+      interval: 1s
+      timeout: 2s
+      retries: 300
+
+  app:
+    depends_on:
+      mysql:
+        condition: service_healthy 
+        restart: true

--- a/docker-compose-postgresql.yml
+++ b/docker-compose-postgresql.yml
@@ -1,0 +1,19 @@
+services:
+  postgresql:
+    image: postgres:12
+    ports:
+      - "5472:5432"
+    env_file:
+      - env.d/development/postgresql
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready"]
+      interval: 1s
+      timeout: 2s
+      retries: 300
+
+  app:
+    depends_on:
+      postgresql:
+        condition: service_healthy 
+        restart: true
+

--- a/docker-compose-sqlite.yml
+++ b/docker-compose-sqlite.yml
@@ -1,0 +1,1 @@
+# This file is empty on propose!

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -1,16 +1,4 @@
 services:
-  postgresql:
-    image: postgres:12
-    ports:
-      - "5472:5432"
-    env_file:
-      - env.d/development/postgresql
-
-  mysql:
-    image: mysql:8.0
-    env_file:
-      - env.d/development/mysql
-    command: --default-authentication-plugin=mysql_native_password
 
   elasticsearch:
     image: "${ELASTICSEARCH_IMAGE:-docker.elastic.co/elasticsearch/elasticsearch:7.14.0}"
@@ -39,8 +27,8 @@ services:
       - ./data/media:/data/media
       - ./data/smedia:/data/smedia
     depends_on:
-      - "${DB_HOST:-postgresql}"
-      - "elasticsearch"
+      elasticsearch:
+        condition: service_started
     stdin_open: true
     tty: true
 

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -5,7 +5,11 @@ services:
     environment:
       - discovery.type=single-node
       - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
-
+    healthcheck:
+      test: ["CMD", "echo", "''", ">", "/dev/tcp/127.0.0.1/9200", "||", "exit", "1"]
+      interval: 1s
+      timeout: 2s
+      retries: 300
   app:
     build:
       context: .
@@ -29,9 +33,15 @@ services:
     depends_on:
       elasticsearch:
         condition: service_started
+      redis-sentinel:
+        condition: service_started
     stdin_open: true
     tty: true
-
+    healthcheck:
+      test: ["CMD", "echo", "''", ">", "/dev/tcp/127.0.0.1/8000", "||", "exit", "1"]
+      interval: 1s
+      timeout: 2s
+      retries: 3600
   nginx:
     image: nginx
     ports:
@@ -84,12 +94,22 @@ services:
       - redis-replica2
     environment:
       - REDIS_MASTER_HOST=redis-primary
+    healthcheck:
+      test: ["CMD-SHELL", "redis-cli -p 26379  ping | grep PONG"]
+      interval: 1s
+      timeout: 2s
+      retries: 300
 
   redis-primary:
     image: docker.io/bitnami/redis:6.0-debian-10
     environment:
       - ALLOW_EMPTY_PASSWORD=yes
       - REDIS_REPLICATION_MODE=master
+    healthcheck:
+      test: ["CMD-SHELL", "redis-cli -p 6379 ping | grep PONG"]
+      interval: 1s
+      timeout: 2s
+      retries: 300
 
   redis-replica1:
     image: docker.io/bitnami/redis:6.0-debian-10
@@ -99,6 +119,11 @@ services:
       - REDIS_MASTER_HOST=redis-primary
     depends_on:
       - redis-primary
+    healthcheck:
+      test: ["CMD-SHELL", "redis-cli -p 6379 ping | grep PONG"]
+      interval: 1s
+      timeout: 2s
+      retries: 300
 
   redis-replica2:
     image: docker.io/bitnami/redis:6.0-debian-10
@@ -108,9 +133,11 @@ services:
       - REDIS_MASTER_HOST=redis-primary
     depends_on:
       - redis-primary
-
-  dockerize:
-    image: jwilder/dockerize
+    healthcheck:
+      test: ["CMD-SHELL", "redis-cli -p 6379 ping | grep PONG"]
+      interval: 1s
+      timeout: 2s
+      retries: 300
 
 networks:
   lms_outside:

--- a/env.d/development/sqlite
+++ b/env.d/development/sqlite
@@ -1,0 +1,5 @@
+# App database configuration
+DB_NAME=db.sqlite3
+
+DB_HOST=sqlite
+DB_ENGINE=django.db.backends.sqlite3

